src/textual/message_pump.py

```diff
@@ -233,7 +240,7 @@ async def _process_messages(self) -> None:
             except CancelledError:
                 raise
             except Exception as error:
-                self.app.panic(error)
+                self.app.on_exception(error)
```

Comment:
@darrenburns pretty sure this is the cause of the "can't render blah..." error you saw.